woocommerce/woocommerce-ios
# Update screenshot generation tests from recent app changes

For Home Screen M1/M2 screenshots pdnsEh-6N-p2



### Description


To prepare for the Home Screen M1/M2 release, we'd like to update our App Store screenshots. There has been bunch of changes that affect the screenshots test since [the last update](https://github.com/woocommerce/woocommerce-ios/pull/5123) in October 2021, especially on new actions that the mock stores haven't implemented.

Major changes include:

- Implemented necessary actions in `MockAppSettingsActionHandler` and `MockStoresManager` with updated indentation on switch cases. `load*` actions are often needed for UI to work properly
- Previously, the stats only work for one month like in [the first screenshot in the last update](https://github.com/woocommerce/woocommerce-ios/pull/5123). This PR updated how we generate visitor and order stats mock data in `ScreenshotObjectGraph` so that the chart shows data for all 12 months of the current year
- A recent change on the Reviews tab shows the navigation bar button conditionally (https://github.com/woocommerce/woocommerce-ios/issues/4295), while `ReviewsScreen` expects the button to show. This PR updated to add an accessibility identifier to the reviews table view and expects the table view to show instead
- After running the screenshots test a few times, I ran into an error where the reviews table keeps changing data and show many duplicate reviews (you can also see this duplication issue in [the `5-dark-review-list` screenshot in the last update](https://github.com/woocommerce/woocommerce-ios/pull/5123)). This PR fixed the issue by deleting all reviews before saving mock reviews

### Testing instructions


- In Xcode, switch to WooCommerceScreenshots target
- Run all tests in this target --> the screenshots UI test should run (going to the screens in the example screenshots) and pass in the end

### Screenshots


1-dark-order-dashboard | 2-dark-order-list | 3-dark-order-detail | 4-dark-order-search
-- | -- | --  | --
<img width="414" alt="iPhone 11-1-dark-order-dashboard" src="https://user-images.githubusercontent.com/1945542/150455392-56ddd2e6-92d7-4a76-b8f0-7cd32a1d25ee.png"> | <img width="414" alt="iPhone 11-2-dark-order-list" src="https://user-images.githubusercontent.com/1945542/150455396-5f3dab5e-bfbc-414c-b48f-a2634d923898.png"> | <img width="414" alt="iPhone 11-3-dark-order-detail" src="https://user-images.githubusercontent.com/1945542/150455398-ad30cf0a-da5a-49e8-9f3b-78502a507eb6.png"> | <img width="414" alt="iPhone 11-4-dark-order-search" src="https://user-images.githubusercontent.com/1945542/150455402-01d0518c-0939-467c-bb91-feb692feabc2.png">


5-dark-review-list | 6-dark-review-details | 7-dark-product-list | 8-dark-product-details
-- | -- | --  | --
<img width="414" alt="iPhone 11-5-dark-review-list" src="https://user-images.githubusercontent.com/1945542/150455421-475c5a0e-09d1-476d-a889-6de9cd736f51.png"> | <img width="414" alt="iPhone 11-6-dark-review-details" src="https://user-images.githubusercontent.com/1945542/150455425-d026655d-3ee4-4d7c-9352-70afd45b407f.png"> | <img width="414" alt="iPhone 11-7-dark-product-list" src="https://user-images.githubusercontent.com/1945542/150455431-6496a0b8-cbb5-4ff9-8833-01014922bb1d.png"> | <img width="414" alt="iPhone 11-8-dark-product-details" src="https://user-images.githubusercontent.com/1945542/150455433-d51f3703-d80b-4ac8-b5b2-ae2bcba3480c.png">


---
